herddb-core/src/main/java/herddb/model/Transaction.java

```diff
@@ -235,19 +235,31 @@ public void unregisterUpgradedLocksOnTable(String tableName, LockHandle lock) {
      * @param key
      * @param value     if null this is a DELETE
      */
-    public synchronized void registerRecordUpdate(String tableName, Bytes key, Bytes value, CommitLogResult writeResult) {
+    public synchronized boolean registerRecordUpdate(String tableName, Bytes key, Bytes value, CommitLogResult writeResult) {
         if (updateLastSequenceNumber(writeResult)) {
-            return;
-        }
-        Map<Bytes, Record> ll = changedRecords.get(tableName);
-        if (ll == null) {
-            ll = new HashMap<>();
-            changedRecords.put(tableName, ll);
+            // this return value is not important
+            return false;
         }
-        if (value == null) {
-            ll.remove(key);
+        Map<Bytes, Record> newRecordsForTable = newRecords.get(tableName);
+        if (newRecordsForTable != null && newRecordsForTable.containsKey(key)) {
```

Comment:
Just a consideration: Could it be better with a Map.compute to avoid the double key lookup?

Reply:
> Just a consideration: Could it be better with a Map.compute to avoid the double key lookup?

I am not sure I understand what you mean here.
We have to enter this branch only if the key is present in `newRecordsForTable`
otherwise we have to enter the other branch.

Reply:
Let this be as is